march_hardware/src/IMotionCube.cpp

```diff
@@ -364,6 +376,91 @@ uint32_t IMotionCube::computeSWCheckSum(int& start_address, int& end_address)
                                  delimiter);
 }
 
+int IMotionCube::verifySetup()
+{
+  int start_address = 0, end_address = 0;
+  uint32_t sw_value = this->computeSWCheckSum(start_address, end_address);
+  // set parameters to compute checksum on the imc
+  int checksum_setup =
+      sdo_bit32_write(slaveIndex, 0x2069, 0, end_address * 65536 + start_address);  // Endaddress leftshifted 4 times
+
+  uint32_t imc_value;
+  int value_size = sizeof(imc_value);
+  // read computed checksum on imc
+  int check_sum_read = sdo_bit32_read(slaveIndex, 0x206A, 0, value_size, imc_value);
+
+  if (!(checksum_setup && check_sum_read))
+  {
+    throw error::HardwareException(error::ErrorType::WRITING_INITIAL_SETTINGS_FAILED,
+                                   "Failed checking the checksum on slave: %d", this->slaveIndex);
+  }
+
+  ROS_DEBUG("The .sw checksum is : %d, and the drive checksum is %d", sw_value, imc_value);
+  if (sw_value == imc_value)
+  {
+    return 1;
+  }
+  else
+  {
+    return 0;
+  }
```

Comment:
Why not make this function boolean and do `return sw_value==imc_value` ?

Reply:
see my other comment on this